Update to current upstream PDCLib

thrimbor opened the pull request:
Updates from 8e10fcf7fd3bac2c5640a86fa165e7fa8c411803 to 4f5b4339a48890599d8fb85bfa2fc6f031d47fe6. Main changes are:
- off-by-one fix in test discovered in #10 
- addition of `at_quick_exit` and `quick_exit`
- addition of `_PDCLIB_Noreturn` to function declarations that are supposed to have it
- extension hooks to extend PDCLib with non-standard functionality (required for libc++)

Apart from squashing, the xbox specific portion is untouched (apart from adding a `#include <synchapi.h>` so we can get rid of the WAIT_FAILED redefinition warning).

When this PR is approved, I'll do the usual tagging & force-push procedure.

JayFoxRox commented:
Thanks for the summary - that's helpful.

> When this PR is approved [...]

With these upstream updates I don't care so much for review (as we trust upstream anyway).

What I *do* care about is what future changes might happen in the near future. This allows us to make a decision wether we should update now or in the near future (to avoid wasting time on rebasing / tagging + submodule updates).
If no further upstream plans are known / planned I'm fine with updating the base.

thrimbor commented:
@JayFoxRox I only know of the plan described in https://github.com/DevSolar/pdclib/issues/5#issuecomment-534523724, but I'd still like to have the extension hooks in our fork soon, so I can upstream more of my libc++ prerequisites and then use the hooks to enable them easily.

I just added a minor change to the xbox-specific code: I added a `#include <synchapi.h>` to `platform/xbox/functions/threads/cnd_wait_timedwait.c`, so we can eliminate the duplicated definition of WAIT_FAILED from `xboxkrnl.h` and get rid of the warning.
